tests/test_serialization.py

```diff
@@ -75,6 +102,55 @@ def torch_state_dict_shared_layers() -> Dict[str, "torch.Tensor"]:
         pytest.skip("torch is not available")
 
 
+@pytest.fixture
+def torch_state_dict_shared_layers_tensor_subclass() -> Dict[str, "torch.Tensor"]:
+    try:
+        import torch
+        from torch.testing._internal.two_tensor import TwoTensor
+
+        t = torch.tensor([4])
+        tensor_subclass_tensor = TwoTensor(t, t)
+
+        t = torch.tensor([4])
+        shared_tensor_subclass_tensor = TwoTensor(t, t)
+        return {
+            "layer_1": torch.tensor([4]),
+            "layer_2": torch.tensor([10]),
+            "layer_3": torch.tensor([30]),
+            "layer_4": torch.tensor([2]),
+            "layer_5": torch.tensor([2]),
+            "layer_6": tensor_subclass_tensor,
+            "ts_shared_1": shared_tensor_subclass_tensor,
+            "ts_shared_2": shared_tensor_subclass_tensor,
+        }
+    except ImportError:
+        pytest.skip("torch is not available")
+
+
+@pytest.fixture
+def torch_state_dict_shared_layers() -> Dict[str, "torch.Tensor"]:
+    try:
+        import torch
+        from torch.testing._internal.two_tensor import TwoTensor
+
+        if is_wrapper_tensor_subclass_available():
+            # TODO: need to fix safetensor support for tensor subclasses before we can add this
+            # to test
+            # shared_layer = TwoTensor(torch.tensor([4]), torch.tensor([4]))
```

Comment:
@Wauplin this seem to fail because safetensor does not support wrapper tensor subclass yet, we can enable this when we add the similar support in safetensors

Reply:
Could we just drop the entire test fixture then? If we remove this, I feel that the most interesting part of the test is gone (having two `TwoTensor` is already done in the other fixture).

Also `code quality` action in the CI is complaining because `torch_state_dict_shared_layers` fixture is declared twice. I think it's just better to remove it for now.

Reply:
Actually we don't have to test on `save_torch_state_dict` using `safetensors`. It is possible to simply test with `split_torch_state_dict_into_shards` (needs to be imported) since what we really care about is the grouping of the tensors, not necessarily the serialization -for now at least-. Could you update the fixture and test in that direction?

Also, `torch_state_dict_shared_layers` is already taken as a name for a fixture so code quality is complaining in the CI.